eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -240,8 +223,14 @@ private void resolveConflict(BomDependencyNoVersion dependencyNoVersion) {
                 eligibleVersion = coreDependencyNameToDependency.get(dependencyNoVersion).getVersion();
                 logger.trace(String.format("\tPicking the version used by azure-core - %s:%s", dependencyNoVersion, eligibleVersion));
             } else {
-                eligibleVersion = versionList.get(versionList.size() - 1);
-                logger.trace(String.format("\tPicking the latest version %s:%s", dependencyNoVersion, eligibleVersion));
+                // We check if any of the assemblies were directly added as inputs, if so, we give that preference.
+                if (inputDependencies.containsKey(dependencyNoVersion)) {
```

Comment:
I think here we should use inputDependency only when it matches the expected dependency in `versioning_client.txt`.

Reply:
I think there is a slight confusion on what override means - the override input dependency filename - is a way of overriding what is in the versioning_client.txt or in another words what assembly should be in the BOM.

So, for the analyzer - inputdependencies is the list of potential BOM roots which includes dependencies from versioning_client.txt as well overridden dependencies,